pkg/apis/kubebind/v1alpha1/apiservicebinding_types.go

```diff
@@ -87,8 +87,135 @@ type APIServiceBindingSpec struct {
+	ClaimRejected AcceptablePermissionClaimState = "Rejected"
+)
+
+// PermissionClaim identifies an object by GR and identity hash.
+// Its purpose is to determine the added permissions that a service provider may
+// request and that a consumer may accept and allow the service provider access to.
+//
+// +kubebuilder:validation:XValidation:rule="!(self.create.donate && self.adopt)",message="donate and adopt are mutually exclusive"
+type PermissionClaim struct {
+	GroupResource `json:","`
+
+	Version string `json:"version"`
+
+	// selector selects which resources are affected by this claim.
+	Selector ResourceSelector `json:"selector,omitempty"`
+
+	// required indicates whether the APIServiceBinding will work if this claim is not accepted.
+	Required bool `json:"required"`
+
+	// Global claims global resources for the given group/resource.
+	// This is mutually exclusive with resourceSelector.
+	// +optional
+	Global bool `json:"global,omitempty"`
+
+	// only for owner Provider
+	//
+	// create determines whether the kube-bind konnector will sync matching objects from the
+	// provider side down to the consumer cluster.
+	Create CreateOptions `json:"create"`
+
+	// adopt set to true means that objects created by the consumer are adopted by the provider.
+	// i.e. the provider will become the owner.
+	Adopt bool `json:"adopt"`
+
+	// onConflict determines how the conflicts between objects on the consumer side
+	// will be resolved.
+	OnConflict OnConflictOptions `json:"onConflict,omitempty"`
+
+	// update lists a number of claimed permissions for the provider.
+	// "field" and "preserving" are mutually exclusive.
+	Update UpdateOptions `json:"update"`
+}
+
+type OnConflictOptions struct {
+	// providerOverrides will make the provider override any object that might already exist
+	// in the consumer cluster if it has the same namespaced name as a resource created by the
+	// provider.
+	ProviderOverwrites bool `json:"providerOverrides"`
+
+	// only for owner provider
+	// When recreateWhenConsumerSideDeleted is true the provider will recreate the object
+	// in case the object is missing on the consumer side. Even if the consumer mistakenly or intentionally
+	// deletes the objet, the provider will recreate it. If the field is set as false,
+	// the provider will not recreate the object in case the object is deleted on the RecreateWhenConsumerSideDeleted
+	// side.
+	RecreateWhenConsumerSideDeleted bool `json:"recreateWhenConsumerSideDeleted"`
+}
+
+type CreateOptions struct {
+	// donate set to true means that a newly created object by the provider is immediately owned by hte consumer.
+	// If false, the object stays in ownership of the provider
+	Donate bool `json:"donate"`
+}
+
+type UpdateOptions struct {
+	// fields are the fields owned by the owner of the claim. If the owner sets values of those
+	// fields, they will be synced to the other participant.
+	// Mutually exclusive with preservings.
```

Comment:
would not use the word "owner" here. The claim owner is always the provider, i.e. claims apply to the provider only.

```suggestion
	// fields are a list of JSON Paths describing which parts of an object the provider wants to control in case 
	// the object is owned by the consumer as a whole. This fields cannot be set of provider owned objects
	// are selected in this claim.
	//
	// This field is mutually exclusive with preservings.
```

Reply:
The owner term was taken from the selector. It describes the binding party that, if not overridden, will be the side to create and delete the resource.
My interpretation of "fields" and "preserving" was that:
"fields" is a whitelist for synchronization, meaning everything in "fields" uses the owner of the claimed resource (owner can be consumer or provider) as the source of truth. Everything not in "fields" uses the other side (if the owner is consumer it used provider) as the source of truth.

"excludes" is a blacklist, meaning that fields that are referenced in "excludes" do not use the owner of the resource as the source of truth, and everything not in "excludes" uses the owner as the source of truth.

Reply:
These claims are always from the point of view of the provider. There are two cases:
1. the provider is owner => `preserving` defines exceptions that the provider (here the owner) does not claim to own
2. the consumer is the owner => `fields` defines exceptions that the provider claims to own despite the consumer being the owner.

In other words: the subject of the claim is always the provider. There is no inversion of the subject when the owner changes.